### PR TITLE
feat(Info): Use generic query function

### PR DESCRIPTION
**Pull request by henrikstranneheim**
## Description

Deprecate function `info` and use generic query function instead. Start splitting the large and complex `store/api` module into crud modules.

### Added

- Another BaseHandler as  dataclass
- An UpdateHandler
### Changed

- Implement `_get_query`instead of def `info`
- Refactor tests

### How to prepare for test
- [ ] ssh to hasta (depending on type of change)
- [ ] activate stage: `us`
- [ ] request trailblazer-stage on hasta: `paxa`
- [ ] install on stage:
    ```Shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_trailblazer -t trailblazer -b refactor_sql_user -a
    ```
- [ ] ssh to clinical-db (depending on type of change)
- [ ] install on stage:
`bash /home/proj/production/servers/resources/clinical-db.scilifelab.se/update-trailblazer-ui-stage.sh [THIS-BRANCH-NAME]`

### How to test
- [ ] login to ...
- [ ] do ...

### Expected test outcome
- [ ] check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review
- [ ] tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: trailblazer/store/base.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+from alchy import Query
+from dataclasses import dataclass
+from typing import Type
+
+from trailblazer.store.models import Model
+
+
+@dataclass
+class BaseHandler_2:
```
Comment (seallard) [inline trailblazer/store/base.py:9]: Is this a temporary class that will be refactored soon? I don't think we should have names like this, maybe the existing base handler class can be updated instead?

Comment (henrikstranneheim) [reply trailblazer/store/base.py:9]: Yes, exactly - it is temporary. Sine the current handler in api.py is called Basehandler, but contains everything. Hoping to move that to crud modules and then rename this.

File: trailblazer/store/crud/update.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+
+class UpdateHandler(BaseHandler_2):
+    """Class for updating items in the database."""
+
+    def update_latest_update_date(self) -> None:
+        """Update when the database was last updated."""
+        info: Info = self._get_query(table=Info).first()
+        info.updated_at: datetime = datetime.now()
+        self.commit()
```
Comment (seallard) [inline trailblazer/store/crud/update.py:14]: We should consider when we want to do our commits. Maybe we only want to do it at a top level, meaning that we commit once a request or cli command has been processed successfully.

**Comment by henrikstranneheim** [comment]
@seallard Please take a look again. I made some modifications.

**Review by seallard — APPROVED** [review]
Looks good 👍

**Comment by henrikstranneheim** [comment]
### trailblazer scan ✅ 
```
tracepath       tracepath6      traceroute      traceroute6     tracker         tracker.ak      tracker-client  tracker-server  trailblazer     trap
θ69° [henrik.stranneheim@hasta:~] [S_base] 1m48s $ trailblazer scan
Status in SLURM: 2257WYLT3 - 271877
        id                   step  status  time_limit  time_elapsed              started
0  4826687  2257WYLT3_demultiplex  FAILED  1-12:00:00             0  2023-06-14T16:36:14
Updated status 2257WYLT3 - 271877: failed
Error logging case - acetuna : FileNotFoundError
Error logging case - uphippo : FileNotFoundError
Error logging case - equalbug : FileNotFoundError
Error logging case - crackyak : FileNotFoundError
Error logging case - unitedbeagle : FileNotFoundError
Error logging case - setamoeba : FileNotFoundError
Error logging case - fleetjay : FileNotFoundError
Error logging case - acetuna : FileNotFoundError
slurm_load_jobs error: Invalid job id specified
Error logging case - HYG7YDSXX : CalledProcessError
Error logging case - gentleamoeba : FileNotFoundError
Error logging case - masterimpala : FileNotFoundError
Error logging case - starakita : FileNotFoundError
Error logging case - goodrodent : FileNotFoundError
Error logging case - strongbug : FileNotFoundError
Error logging case - masterwombat : FileNotFoundError
Error logging case - warmcalf : FileNotFoundError
Error logging case - jointstag : FileNotFoundError
Error logging case - finerobin : FileNotFoundError
Error logging case - wantedhen : FileNotFoundError
Error logging case - cleanleech : PermissionError
Error logging case - socialearwig : FileNotFoundError
Error logging case - topsdodo : FileNotFoundError
Error logging case - usablefeline : PermissionError
Error logging case - 233471 : FileNotFoundError
Error logging case - crispshiner : FileNotFoundError
Error logging case - supertiger : PermissionError
Error logging case - fullfrog : FileNotFoundError
Error logging case - toplouse : FileNotFoundError
Error logging case - sunnycondor : FileNotFoundError
Error logging case - keyeft : FileNotFoundError
Error logging case - robustalien : PermissionError
Error logging case - busyowl : PermissionError
Error logging case - boldhound : PermissionError
Error logging case - vastboar : FileNotFoundError
Error logging case - topsworm : PermissionError
Error logging case - setcolt : PermissionError
Error logging case - proudfish : PermissionError
Error logging case - funnyaphid : FileNotFoundError
Error logging case - onestag : PermissionError
Error logging case - newgopher : PermissionError
Error logging case - helpedlab : EmptySqueueError
Error logging case - goodbengal : EmptySqueueError
Error logging case - grandaphid : PermissionError
All analyses updated!
```

**Comment by henrikstranneheim** [comment]
Closing this continuing here instead: https://github.com/Clinical-Genomics/trailblazer/pull/218

**Pull request closed without merging**